### PR TITLE
tournament  page with table

### PR DESCRIPTION
**Pull request by John-A-J**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -14,7 +14,16 @@ const config = {
 const mustacheExpress = require('mustache-express')
 const app = express();
 let date = new Date();
+const mysql = require('mysql');
 
+const moment = require('moment')
+
+
+function parsedate(date){
```
Comment (rcdenyer) [inline app/app.js]: would be better with a capital D

File: app/app.js [hunk 2]
```diff
@@ -14,7 +14,16 @@ const config = {
...
+	let formattedDate = moment(date).format('dddd MMMM Do, YYYY')
+	console.log(formattedDate)
```
Comment (rcdenyer) [inline app/app.js]: delete plz

File: app/app.js [hunk 3]
```diff
@@ -79,6 +89,25 @@ function showArchersPage(req, res){
 	})
 }
 
+function showTournamentsPage(req, res){
+	executeQuery('SELECT * FROM `tournament` WHERE datetime_end > now() ORDER BY datetime_end', (result) =>{
```
Comment (rcdenyer) [inline app/app.js]: perhaps instead of * list the keys you want to use as may add extra columns

File: app/app.js [hunk 4]
```diff
@@ -79,6 +89,25 @@ function showArchersPage(req, res){
...
+
+		result.forEach((row)=> {
+
+			row.datetime_start = parsedate(row.datetime_start)
+			row.datetime_end  = parsedate(row.datetime_end)
+
+			formattedResults.push(row)
+		})
+
+		app.render('content_tournament.html', {tournament_result:formattedResults}, (err,content)=>{
```
Comment (rcdenyer) [inline app/app.js]: maybe tournament-list for consistency with archers

File: app/app.js [hunk 5]
```diff
@@ -14,7 +14,16 @@ const config = {
 const mustacheExpress = require('mustache-express')
 const app = express();
 let date = new Date();
```
Comment (rcdenyer) [inline app/app.js]: needs deleting see below for use

File: app/app.js [hunk 6]
```diff
@@ -79,6 +89,25 @@ function showArchersPage(req, res){
...
+		result.forEach((row)=> {
+
+			row.datetime_start = parsedate(row.datetime_start)
+			row.datetime_end  = parsedate(row.datetime_end)
+
+			formattedResults.push(row)
+		})
+
+		app.render('content_tournament.html', {tournament_result:formattedResults}, (err,content)=>{
+			res.render('fullpage.html', {title:"Welcome to IWAO", year:date.getFullYear(), content: content})
```
Comment (rcdenyer) [inline app/app.js]: remove year and hardcode 2017 for now

File: app/app.js (hunk 5 above)
Comment (ivorcheung) [reply app/app.js]: Removed

File: app/app.js (hunk 1 above)
Comment (ivorcheung) [reply app/app.js]: Changed

File: app/app.js [hunk 7]
```diff
@@ -14,7 +14,16 @@ const config = {
 const mustacheExpress = require('mustache-express')
 const app = express();
 let date = new Date();
+const mysql = require('mysql');
 
+const moment = require('moment')
+
+
+function parsedate(date){
+	let formattedDate = moment(date).format('dddd MMMM Do, YYYY')
```
[the remark this answers is not in the document]
Comment (ivorcheung) [reply app/app.js]: Done

File: app/app.js (hunk 2 above)
Comment (ivorcheung) [reply app/app.js]: Done

File: app/app.js (hunk 3 above)
Comment (ivorcheung) [reply app/app.js]: Changed the SQL Statement for better readability

File: app/app.js (hunk 4 above)
Comment (ivorcheung) [reply app/app.js]: Done

File: app/app.js (hunk 6 above)
Comment (ivorcheung) [reply app/app.js]: Done

**Pull request merged**